Major spec cleanup

lamawithonel opened the pull request:
I was going to add some new functionality, but the spec tests were a little hard to follow and a lot of the platform-dependent tests only included contexts for RedHat.  I ended up doing a pretty massive overhaul and this is the result.  Most of the work is in aeb57a5, the other commits are just in support of it.  The coverage should be just as good, if not better than it was before.  It also includes some changes for RuboCop.  Let me know if you want anything changed.

jhoblitt commented:
Awesome!  I will take a look after travis finishes.

jhoblitt commented on Gemfile:
```diff
@@ -1,25 +1,20 @@
-  gem 'puppet', :require => false
-end
+ENV['PUPPET_GEM_VERSION'] && puppetversion = ENV['PUPPET_GEM_VERSION']
+gem 'puppet', puppetversion, :require => false
```
Why not compress this down to a one-liner?

jhoblitt commented on spec/unit/classes/smartd_spec.rb:
```diff
@@ -1,541 +1,414 @@
+      when 'RedHat'
+        service_name = 'smartd'
+        smartd_conf_path = case facts[:operatingsystem]
+                           when 'RedHat', 'CentOS', 'Scientific', 'OracleLinux', 'SLC', 'OEL'
```
Does rubocop want this level of indentation?

lamawithonel replied:
It does.  When a case statement is used to assign only one variable it wants the `var = case..` syntax, and it wants the keywords of the case statement to align.  I can disable the check if you want.  I'm not particularly fond of it myself, though I prefer to follow the defaults unless there's a good reason to differ.

jhoblitt commented on .rubocop.yml:
```diff
@@ -0,0 +1,26 @@
+---
```
This is great.  How about pulling in the rakefile target and adding it to travis as well?

lamawithonel replied:
I tried that initially, but it's incompatible with Ruby 1.8 and Puppet's vendor-bundled SafeYAML module.  I've added it again with some conditional logic restricting it to Ruby 2.2 and newer.  It's not as clean as the rest of the checks, but it passes the latest build.

jhoblitt commented on Gemfile:
```diff
@@ -1,25 +1,20 @@
   gem 'puppet-syntax',            :require => false
   gem 'rspec-puppet', '~> 2.2',   :require => false
+  gem 'rspec-puppet-facts',       :require => false
   gem 'metadata-json-lint',       :require => false
```
rubocop should be added to the Gemfile.

jhoblitt commented:
Whoa!  This is fantastic.  I've made a few minor comments.

lamawithonel commented:
Have you had a chance to review this?  I addressed the issues you brought up, but I've been away for a few weeks.  I'm now using it in production with #52 if it makes any difference.

lamawithonel commented:
The last merge made most of my previous changes really difficult to rebase and I pretty much have to rewrite it.  For now I'm going to scale this PR back so it can get merged more quickly.  I will open other PRs for the other changes.  This PR now just covers the Transpec deprecation cleanup and RuboCop support.

jhoblitt commented:
Sorry for being asleep at the wheel here!  I'm going to merge this before I tackle the rspec rot.  Thanks!
